File: services/sync-base/app/utilities/random.ts
export interface Foo {
  id: string;
  name: string;
  description: string | null;
  status: FooStatus;
  priority: number;
  isActive: boolean;
  metadata: Record<string, any>;
  config: object;
  tags: string[];
  score: number;
  largeText: string;
  createdAt: Date;
  updatedAt: Date;
}

export enum FooStatus {
  ACTIVE = "active",
  INACTIVE = "inactive",
  PENDING = "pending",
  ARCHIVED = "archived",
}

function randomString(length: number) {
  return Math.random().toString(36).substring(2, 2 + length);
}

function randomEnum<T extends object>(anEnum: T): T[keyof T] {
  const values = Object.values(anEnum) as T[keyof T][];
  return values[Math.floor(Math.random() * values.length)];
}

function randomDate() {
  return new Date(Date.now() - Math.floor(Math.random() * 1000000000));
}

export function randomFoo(): Foo {
  return {
    id: crypto.randomUUID(),
    name: randomString(8),
    description: Math.random() > 0.5 ? randomString(20) : null,
    status: randomEnum(FooStatus),
    priority: Math.floor(Math.random() * 10),
    isActive: Math.random() > 0.5,
    metadata: { foo: randomString(5), bar: Math.random() },
    config: { setting: randomString(4) },
    tags: Array.from({ length: Math.floor(Math.random() * 5) + 1 }, () => randomString(4)),
    score: parseFloat((Math.random() * 100).toFixed(2)),
    largeText: randomString(100),
    createdAt: randomDate(),
    updatedAt: randomDate(),
  };
}

File: services/data-warehouse/app/connectors/random.py
from typing import List
from datetime import datetime, timedelta
import random
import string
import uuid
from app.ingest.models import Foo, FooStatus

def random_string(length: int) -> str:
    return ''.join(random.choices(string.ascii_lowercase + string.digits, k=length))

def random_enum(enum_class: type[FooStatus]) -> FooStatus:
    return random.choice(list(enum_class))

def random_date() -> datetime:
    seconds_ago = random.randint(0, 1000000000)
    return datetime.now() - timedelta(seconds=seconds_ago)

def random_foo() -> Foo:
    return Foo(
        id=str(uuid.uuid4()),
        name=random_string(8),
        description=random_string(20) if random.random() > 0.5 else None,
        status=random_enum(FooStatus),
        priority=random.randint(0, 9),
        is_active=random.random() > 0.5,
        tags=[random_string(4) for _ in range(random.randint(1, 5))],
        score=round(random.random() * 100, 2),
        large_text=random_string(100),
        created_at=random_date(),
        updated_at=random_date()
    )